staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt-config.go

```diff
@@ -0,0 +1,284 @@
+			close(plState.emptyReturns)
+		} else {
+			if plState.emptyHandler == nil {
+				klog.V(3).Infof("Priority level %s became undesired", plName)
+				plState.emptyHandler = &emptyRelay{reqMgmt, plState}
+				newlyQuiescent = append(newlyQuiescent, plState)
+			}
+			newRMS.priorityLevelStates[plName] = plState
+			if !plState.config.Exempt {
+				shareSum += float64(plState.config.AssuredConcurrencyShares)
+			}
+		}
+	}
+	for _, plState := range newRMS.priorityLevelStates {
+		if plState.config.Exempt {
+			continue
+		}
+		plState.concurrencyLimit = int(math.Ceil(float64(reqMgmt.serverConcurrencyLimit) * float64(plState.config.AssuredConcurrencyShares) / shareSum))
+		if plState.fqs == nil {
+			plState.fqs = reqMgmt.fairQueuingFactory.NewFairQueuingSystem(plState.concurrencyLimit, int(plState.config.Queues), int(plState.config.QueueLengthLimit), reqMgmt.requestWaitLimit, reqMgmt.clock)
+		} else {
+			plState.fqs.SetConfiguration(plState.concurrencyLimit, int(plState.config.Queues), int(plState.config.QueueLengthLimit), reqMgmt.requestWaitLimit)
+		}
+	}
+	reqMgmt.curState.Store(newRMS)
+	// We do the following only after updating curState to guarantee
+	// that if Wait returns `quiescent==true` then a fresh load from
+	// curState will yield an RMState that is at least as up-to-date
+	// as the data here.
+	for _, plState := range newlyQuiescent {
+		plState.fqs.Quiesce(plState.emptyHandler)
+	}
+	return true
+}
+
+// currentEmptyReturned pull the empty returns to see if the current
+// one is there
+func (pls *PriorityLevelState) currentEmptyReturned() bool {
+	for {
+		select {
+		case eh := <-pls.emptyReturns:
+			if eh == pls.emptyHandler {
+				return true
+			}
+		default:
+			return false
+		}
+	}
+}
+
+func applyPriorityLevelSpecDefaults(plName string, spec *rmtypesv1a1.PriorityLevelConfigurationSpec) {
+	if spec.AssuredConcurrencyShares < 1 {
+		klog.Errorf("Found AssuredConcurrencyShares=%v for priority level %s, fixing it to 100", spec.AssuredConcurrencyShares, plName)
+		spec.AssuredConcurrencyShares = 100
+	}
+	if spec.Queues < 1 {
+		klog.Errorf("Fixing number of queues for priority level %s to be 1", plName)
+		spec.Queues = 1
+	} else if spec.Queues > 65536 {
+		klog.Errorf("Reducing number of queues for priority level %s from %v to %v", plName, spec.Queues, 1024)
+		spec.Queues = 1024
+	}
+	if spec.HandSize < 1 {
+		klog.V(3).Infof("Defaulting hand size for priority level %s to 1", plName)
+		spec.HandSize = 1
+	}
+}
+
+func warnFlowSchemaSpec(fsName string, spec *rmtypesv1a1.FlowSchemaSpec, newPriorities, oldPriorities map[string]*PriorityLevelState) bool {
+	plName := spec.PriorityLevelConfiguration.Name
+	if newPriorities[plName] == nil {
+		problem := "non-existent"
+		if oldPriorities[plName] != nil {
+			problem = "undesired"
+		}
+		klog.Warningf("FlowSchema %s references %s priority level %s and will thus not match any requests", fsName, problem, plName)
```

Comment:
i think we should probably set a condition in the status subresource of the PL in this case? we can do it in a follow-up

Reply:
Remember there are multiple apiservers.  I am dubious about each posting status.